src/uucore/src/lib/mods/locale.rs

```diff
@@ -0,0 +1,330 @@
+
+// A struct to handle localization
+struct Localizer {
+    bundle: FluentBundle<FluentResource>,
+}
+
+impl Localizer {
+    fn new(bundle: FluentBundle<FluentResource>) -> Self {
+        Self { bundle }
+    }
+
+    fn format(&self, id: &str, args: Option<&FluentArgs>, default: &str) -> String {
+        match self.bundle.get_message(id).and_then(|m| m.value()) {
+            Some(value) => {
+                let mut errs = Vec::new();
+                self.bundle
+                    .format_pattern(value, args, &mut errs)
+                    .to_string()
+            }
+            None => default.to_string(),
+        }
+    }
+}
+
+// Global localizer stored in thread-local OnceLock
+thread_local! {
+    static LOCALIZER: OnceLock<Localizer> = const { OnceLock::new() };
+}
+
+// Initialize localization with a specific locale and config
+fn init_localization(
+    locale: &LanguageIdentifier,
+    config: &LocalizationConfig,
+) -> Result<(), LocalizationError> {
+    let bundle = create_bundle(locale, config)?;
+    LOCALIZER.with(|lock| {
+        let loc = Localizer::new(bundle);
+        lock.set(loc)
+            .map_err(|_| LocalizationError::Bundle("Localizer already initialized".into()))
+    })?;
+    Ok(())
+}
+
+// Create a bundle for a locale with fallback chain
+fn create_bundle(
+    locale: &LanguageIdentifier,
+    config: &LocalizationConfig,
+) -> Result<FluentBundle<FluentResource>, LocalizationError> {
+    // Create a new bundle with requested locale
+    let mut bundle = FluentBundle::new(vec![locale.clone()]);
+
+    // Try to load the requested locale
+    let mut locales_to_try = vec![locale.clone()];
+    locales_to_try.extend_from_slice(&config.fallback_locales);
+
+    // Always ensure DEFAULT_LOCALE is in the fallback chain
+    let default_locale: LanguageIdentifier = DEFAULT_LOCALE
+        .parse()
+        .map_err(|_| LocalizationError::Parse("Failed to parse default locale".into()))?;
+
+    if !locales_to_try.contains(&default_locale) {
+        locales_to_try.push(default_locale);
+    }
+
+    // Try each locale in the chain
+    let mut loaded = false;
+    let mut tried_paths = Vec::new();
+
+    for try_locale in locales_to_try {
+        let locale_path = config.get_locale_path(&try_locale);
+        tried_paths.push(locale_path.clone());
+
+        if let Ok(ftl_string) = fs::read_to_string(&locale_path) {
+            let resource = FluentResource::try_new(ftl_string).map_err(|_| {
+                LocalizationError::Parse(format!(
+                    "Failed to parse localization resource for {}",
+                    try_locale
+                ))
+            })?;
+
+            bundle.add_resource(resource).map_err(|_| {
+                LocalizationError::Bundle(format!(
+                    "Failed to add resource to bundle for {}",
+                    try_locale
+                ))
+            })?;
+
+            loaded = true;
+            break;
```

Comment:
Instead of a `break` you could do an early return. It would make the `loaded` variable obsolete.